lib/core/tree.js

```diff
@@ -0,0 +1,56 @@
+const { wellknownHeaderNames } = require('./constants')
+
+class Tree {
+  #node = {}
+  #depth = 0
+
+  /**
+   * Lowercases key and inserts its value into the node.
+   * @param {string} value
+   */
+  insert (value) {
+    const target = Buffer.from((value = value.toLowerCase()))
+    const length = target.length
+    let node = this.#node
+    for (let i = 0; i < length; ++i) {
+      const key = target[i]
+      node[key] ??= {}
```

Comment:
Does using an array here make any difference?

Reply:
Array space must be allocated.

Reply:
```
node[key] ??= new Array(257)
```